### PR TITLE
29 add csv ingestion for simple protocols

### PR DESCRIPTION
**Pull request by andySigler**
This branch makes some minor changes to the front end:
1. Added feature for ingesting CSV files, which are then parsed into our JSON format. It expects only a p20 pipette, and runs Transfer commands with a droptip after each transfer. This feature should probably not be developed any further.
2. Prevents users from triggering smoothie-related UI buttons if the serial port is not connected
3. Pressing Home will now also erase the protocol from the backend (requested by testers)
4. The Run button is re-enabled after protocol finished or is cancelled (bug fix)

### CODE REVIEW

**Comment by SimplyAhmazing** [comment]
Approving this for the sale..

File: src/js/loadFiles.js [hunk 1]
```diff
@@ -244,9 +244,11 @@ function loadFile(e) {
 
   if(files[0]){
     var _F = files[0];
-    TIPRACKS = {'a':[],'b':[]}; // clear tipracks
+
     _FILENAME = _F.name;
 
+    console.log(_FILENAME);
```
Comment (levthedev) [inline src/js/loadFiles.js:250]: Is this supposed to be here?

File: src/js/loadFiles.js [hunk 2]
```diff
@@ -538,3 +556,123 @@ function getAJAX(filepath,callback) {
...
+    }
+
+    var new_protocol = {
+      'deck' : current_deck,
+      'head' : current_head,
+      'ingredients' : current_ingredients,
+      'instructions' : current_instructions
+    }
+
+    //console.log(JSON.stringify(new_protocol,undefined,2));
```
Comment (levthedev) [inline src/js/loadFiles.js:670]: See above comment

File: src/js/loadFiles.js (hunk 1 above)
Comment (andySigler) [inline src/js/loadFiles.js:250]: Shouldn't be, but the front-end is filled with `console.log`'s so my eyes weren't looking for these

File: src/js/loadFiles.js (hunk 2 above)
Comment (andySigler) [inline src/js/loadFiles.js:670]: Another ticket could be made for cleaning up the front-end JS code

**Pull request merged**